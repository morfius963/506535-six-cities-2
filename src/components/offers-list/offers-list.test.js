import React from "react";
import renderer from "react-test-renderer";
import OffersList from "./offers-list.jsx";
import OfferCard from "../offer-card/offer-card.jsx";
import fixtureData from "../../__fixtures__/offers.js";

jest.mock(`../offer-card/offer-card.jsx`, () => jest.fn().mockReturnValue(null));

describe(`snapshot test`, () => {
  it(`Component correctly renders`, () => {
    const tree = renderer
      .create(
          <OffersList
            offers = {fixtureData}
          />
      )
      .toJSON();
    expect(OfferCard).toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });
});
