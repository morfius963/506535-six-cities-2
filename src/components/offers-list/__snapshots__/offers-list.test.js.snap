// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test Component correctly renders 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container container"
  >
    <section
      className="cities__places places"
    >
      <h2
        className="visually-hidden"
      >
        Places
      </h2>
      <b
        className="places__found"
      >
        312 places to stay in Amsterdam
      </b>
      <form
        action="#"
        className="places__sorting"
        method="get"
      >
        <span
          className="places__sorting-caption"
        >
          Sort by
        </span>
        <span
          className="places__sorting-type"
          tabIndex="0"
        >
          Popular
          <svg
            className="places__sorting-arrow"
            height="4"
            width="7"
          >
            <use
              xlinkHref="#icon-arrow-select"
            />
          </svg>
        </span>
        <ul
          className="places__options places__options--custom places__options--opened"
        >
          <li
            className="places__option places__option--active"
            tabIndex="0"
          >
            Popular
          </li>
          <li
            className="places__option"
            tabIndex="0"
          >
            Price: low to high
          </li>
          <li
            className="places__option"
            tabIndex="0"
          >
            Price: high to low
          </li>
          <li
            className="places__option"
            tabIndex="0"
          >
            Top rated first
          </li>
        </ul>
      </form>
      <div
        className="cities__places-list places__list tabs__content"
      />
    </section>
    <div
      className="cities__right-section"
    />
  </div>
</div>
`;
